src/Application/ApplicationFileProcessor.php

```diff
@@ -55,6 +56,15 @@ public function run(Configuration $configuration, InputInterface $input): Proces
     {
         $filePaths = $this->fileFactory->findFilesInPaths($configuration->getPaths(), $configuration);
 
+        if ($this->containsVendorPath($filePaths)) {
```

Comment:
This cause warning on downgrade:

https://github.com/rectorphp/rector-src/actions/runs/7714674283/job/21027467954#step:10:13

which should be ok, I think this should be on documentation instead, also, it make performance downgrade as it add loops.

Reply:
Good point :+1:  I'll add contra-check with downgrade rules.

I think the documentation mentions this, but dev don't read documentation :) having a warning right in the tool miss-use allows to correct it right at that moment

Reply:
Ref https://github.com/rectorphp/rector-src/pull/5526